ed_sensor_integration/include/ed/laser/updater.h

```diff
@@ -0,0 +1,120 @@
+#ifndef ED_LASER_LASER_UPDATER_H_
+#define ED_LASER_LASER_UPDATER_H_
+
+#include "ed/laser/entity_update.h"
+
+#include <ed/convex_hull.h>
+#include <ed/init_data.h>
+#include <ed/world_model.h>
+
+#include <geolib/sensors/LaserRangeFinder.h>
+
+#include <map>
+#include <string>
+
+typedef std::vector<unsigned int> ScanSegment;
+
+class LaserUpdater
+{
+
+public:
+    /**
+     *  @brief Configure updater
+     *
+     *  @param[in] config Configuration which may contain the following fields:
+     *      world_association_distance
+     *      min_segment_size_pixels
+     *      segment_depth_threshold
+     *      min_cluster_size
+     *      max_cluster_size
+     *      max_gap_size
+     *      fit_entities Whether or not to fit an entity (0 or 1)
+     */
+    void configure(tue::Configuration& config);
+
+    /**
+     * @brief update update the worldmodel based on a novel laserscan message.
+     *
+     * @param[in] world worldmodel to be updated
+     * @param[in] scan laserscan message
+     * @param[in] sensor_pose pose of the sensor at the time of the measurement
+     * @param[out] req update request
+     */
+    void update(const ed::WorldModel& world, std::vector<double>& sensor_ranges,
+                const geo::Pose3D& sensor_pose, const double timestamp, ed::UpdateRequest& req);
+
+    /**
+     * @brief configure the LRF model based on a laserscan message
+     *
+     * @param num_beams number of beams in the lrf model
+     * @param angle_min angle corresponding to the first beam in radians
+     * @param angle_max angle corresponding to the final beam in radians
+     * @param range_min minimum distance that can be detected with the lrf in meters
+     * @param range_max maximum distance that can be detected with the lrf in meters
+     */
+    void configureLaserModel(uint num_beams, double angle_min, double angle_max, double range_min, double range_max)
+    {
+        lrf_model_.setNumBeams(num_beams);
+        lrf_model_.setAngleLimits(angle_min, angle_max);
+        lrf_model_.setRangeLimits(range_min, range_max);
+    }
+
+     /**
+     * @brief set the frame of the laser model
+     *
+     * @param frame_id frame id of the laser
+     */
+    inline void setLaserFrame(std::string frame_id) { lrf_frame_ = frame_id; }
```

Comment:
```suggestion
    inline void setLaserFrame(const std::string& frame_id) { lrf_frame_ = frame_id; }
```

Reply:
does this make any difference when it is an inline function?